### PR TITLE
Implement missing LRF 4:2:2 support

### LINKED ISSUE #1692: assertion failed: units_x + units_cols <= rs_cols + 1 with ChromaSampling::Cs422
This is a regression from @xiphmont 's commit 70cb11ff76e553752c9963d974afb3fbc4d3375b some days ago. Before that it worked fine, afterwards not.

The following change to `simple_encoding.rs` shows the assertion

```diff
diff --git a/examples/simple_encoding.rs b/examples/simple_encoding.rs
index d9ba817..f038e06 100644
--- a/examples/simple_encoding.rs
+++ b/examples/simple_encoding.rs
@@ -5,10 +5,11 @@ use rav1e::*;
 fn main() {
   let mut cfg = Config::default();
 
-  cfg.enc.width = 64;
-  cfg.enc.height = 96;
+  cfg.enc.width = 160;
+  cfg.enc.height = 120;
 
   cfg.enc.speed_settings = SpeedSettings::from_preset(9);
+  cfg.enc.chroma_sampling = color::ChromaSampling::Cs422;
 
   let mut ctx: Context<u16> = cfg.new_context().unwrap();
 
```

The width/height change also is relevant, but not sure what the relationship there is.

Fails with

```
Sending frame 0
Sending frame 1
Sending frame 2
Sending frame 3
Sending frame 4
Sending frame 5
Sending frame 6
Sending frame 7
Sending frame 8
Sending frame 9
Sending frame 10
thread 'main' panicked at 'assertion failed: units_x + units_cols <= rs_cols + 1', src/tiling/tile_restoration_state.rs:385:1
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:200
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:214
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:477
   7: std::panicking::begin_panic
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/libstd/panicking.rs:411
   8: rav1e::tiling::tile_restoration_state::TileRestorationStateMut::new
             at ./src/tiling/tile_state.rs:0
   9: rav1e::tiling::tile_state::TileStateMut<T>::new
             at ./src/tiling/tile_state.rs:113
  10: <rav1e::tiling::tiler::TileContextIterMut<T> as core::iter::traits::iterator::Iterator>::next
             at ./src/tiling/tiler.rs:190
  11: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/liballoc/vec.rs:1883
  12: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/liballoc/vec.rs:1796
  13: core::iter::traits::iterator::Iterator::collect
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b/src/libcore/iter/traits/iterator.rs:1466
  14: rav1e::api::internal::ContextInner<T>::compute_lookahead_motion_vectors
             at ./src/api/internal.rs:575
  15: rav1e::api::internal::ContextInner<T>::compute_lookahead_data
             at ./src/api/internal.rs:792
  16: rav1e::api::internal::ContextInner<T>::send_frame
             at ./src/api/internal.rs:292
  17: rav1e::api::Context<T>::send_frame
             at ./src/api/mod.rs:320
  18: simple_encoding::main
             at examples/simple_encoding.rs:22
```

### PR DESCRIPTION
**Pull request by xiphmont**
Restoration (or more specifically the LRU partition tracking code) did not implement support for the rectangular superblocks used by 4:2:2 inputs.  This implements the missing support and fixes #1692.

AWCY test metrics incoming.

### CODE REVIEW

File: src/rdo.rs [hunk 1]
```diff
@@ -1648,30 +1650,38 @@ pub fn rdo_loop_decision<T: Pixel>(
...
   // Determine area of optimization: Which plane has the largest LRUs?
   // How many LRUs for each?
-  let mut sb_wh = 1; // how many superblocks wide the largest LRU
-                     // is/how much we're processing (same thing)
-  let mut lru_wh = [0; PLANES]; // how manu LRUs we're processing
+  let mut sb_w = 1; // how many superblocks wide the largest LRU
+                    // is/how much we're processing (same thing)
+  let mut sb_h = 1; // how many superblocks wide the largest LRU
+                    // is/how much we're processing (same thing)
+  let mut lru_w = [0; PLANES]; // how manu LRUs we're processing
```
Comment (YaLTeR) [inline src/rdo.rs]: nit, also next line
```suggestion
  let mut lru_w = [0; PLANES]; // how many LRUs we're processing
```

**Comment by xiphmont** [comment]
Oh, yeah, I need to do the rustfmt thing as yet.  I wasn't expecting a
review that fast :-)
Until AWCY comes in, this is still 'in testing'.

**Comment by xiphmont** [comment]
AWCY shows no change on objective-1-fast (as expected).
Do we have any 4:2:2 test sets?  The test in #1692 now passes as do a few 4:2:2 clips here, but that's not much of a test.

**Comment by lu-zero** [comment]
Some tests seem to suffer.

**Comment by xiphmont** [comment]
yes, fixing
Monty

**Pull request merged**